### PR TITLE
Add --action-tag-filter option for selective action processing

### PR DESCRIPTION
**Pull request by kkaarreell**
Implement support for filtering actions by tags using --action-tag-filter option. This enables splitting test execution into logical parts based on action categorization (e.g., tier1, tier2, security, performance).

Key features:
- Add action_tags field to IssueAction and Issue models
- Implement --action-tag-filter CLI option with regex pattern matching
- Support "ANY" tag matching logic (action included if any tag matches)
- Automatically include parent actions when children match filters
- Combine with existing filters (--action-id-filter, --issue-id-filter) using AND logic (intersection)
- Override auto_schedule=False when filters are specified
- Display action tags in list command output

Implementation details:
- Created _build_action_tag_filtered_list() helper function
- Created _build_combined_action_filtered_list() to handle filter intersection logic consistently across all stages
- Added _should_filter_by_action_tags() method to CLIContext
- Applied filtering at YAML loading and issue-config processing stages
- Ensured consistent filter behavior across jira, schedule, execute, report, and list commands

Testing:
- 40 new unit tests covering all filtering scenarios
- Tests for single filters, combined filters, parent inclusion, pattern matching, and edge cases
- All 192 unit tests passing

Documentation:
- Updated README.md with detailed --action-tag-filter usage examples
- Added action_tags field documentation in issue-config section

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add tag-based action filtering across CLI, models, and scheduling to enable selective processing of tagged actions, including parent actions, and document the new behavior.

New Features:
- Introduce action_tags on Issue and IssueAction to categorize actions and surface them in generated jobs and list output.
- Add --action-tag-filter CLI option with a regex-based expression language (AND/OR/NOT) to selectively process actions by tags across all subcommands.
- Implement combined action ID and tag filtering with intersection semantics, ensuring parent and ancestor actions of matching children are included.
- Allow tag-based filters to override recipe auto_schedule=False so explicitly filtered actions are still scheduled.

Enhancements:
- Unify action filtering logic via shared helpers and a combined filtered-list builder, simplifying how filters are applied during YAML loading and issue-config processing.
- Improve logging messages around filtered actions to reflect generic user-provided filters rather than specific flag names.

Documentation:
- Extend README with documentation and examples for action_tags in issue-config and the new --action-tag-filter option, including expression syntax and usage patterns.

Tests:
- Add comprehensive unit tests for tag filter parsing/evaluation, tag-based YAML/job filtering, combined action-id and action-tag filtering, and CLIContext tag filtering behavior.

### CODE REVIEW

**Comment by kkaarreell** [comment]
@sourcery-ai review

File: tests/unit/test_action_tag_filter_cli.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        assert result is False
+
+        # File with non-matching action_id but matching tag - should be filtered
+        result2 = _should_filter_yaml_file(
+            temp_yaml_files['with_other_tags'],
+            action_id_pattern, None, None, tag_pattern, mock_logger)
+        assert result2 is True
+
+
+class TestActionTagFilterWithCLIContext:
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply tests/unit/test_action_tag_filter_cli.py:155]: I think the existing tests are sufficient.

File: newa/models/settings.py [hunk 2]
```diff
@@ -495,6 +502,43 @@ def _should_filter_by_issue_id(
             "regular expression.")
         return False
 
+    def _should_filter_by_action_tags(
+            self,
+            action_tags: Optional[list[str]],
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply newa/models/settings.py:508]: fixed

File: tests/unit/test_jira_summary_and_schedule.py [hunk 3]
```diff
@@ -390,7 +390,8 @@ def test_schedule_command_skips_auto_schedule_false_without_filters(self, mock_c
         # Verify skip log message
         mock_ctx.logger.info.assert_called_with(
             f'Skipping jira job {JIRA_NONE_ID}-123 - auto_schedule is disabled. '
-            f'Use --schedule-all or --action-id-filter/--issue-id-filter to override.',
+            f'Use --schedule-all or --action-id-filter/--issue-id-filter/'
+            f'--action-tag-filter to override.',
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply tests/unit/test_jira_summary_and_schedule.py:394]: added test_schedule_command_processes_auto_schedule_false_with_tag_filter

File: tests/unit/test_filters.py [hunk 4]
```diff
@@ -417,3 +417,197 @@ def test_should_filter_wrong_artifact_type(self, mock_logger):
...
+            ['performance', 'tier2', 'nightly'])
+        assert result is False
+        # Check debug log was called
+        ctx_with_tag_filter.logger.debug.assert_called_once()
+
+    def test_non_matching_tags_returns_true(self, ctx_with_tag_filter):
+        """When no tags match pattern, should filter (return True)."""
+        result = ctx_with_tag_filter._should_filter_by_action_tags(
+            ['performance', 'nightly'])
+        assert result is True
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply tests/unit/test_filters.py:473]: fixed

**Comment by kkaarreell** [comment]
@sourcery-ai review

File: newa/cli/main.py [hunk 5]
```diff
@@ -79,6 +85,14 @@ def _should_filter_yaml_file(
                     f'does not match filter)')
                 return True
 
+        # Check action_tag_filter if specified
+        if action_tag_filter:
+            action_tags = yaml_data.get('jira', {}).get('action_tags')
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply newa/cli/main.py:90]: fixed
